mobile/test/common/mocks/common/mocks.h

```diff
@@ -0,0 +1,52 @@
+#include "gmock/gmock.h"
+#include "library/common/common/system_helper.h"
+
+namespace Envoy {
+namespace test {
+
+// Mock implementation of SystemHelper.
+class MockSystemHelper : public SystemHelper {
+public:
+  MockSystemHelper();
+
+  // SystemHelper:
+  MOCK_METHOD(bool, isCleartextPermitted, (absl::string_view hostname));
+  MOCK_METHOD(envoy_cert_validation_result, validateCertificateChain,
+              (const std::vector<std::string>& certs, absl::string_view hostname));
+  MOCK_METHOD(void, cleanupAfterCertificateValidation, ());
+};
+
+// SystemHelperPeer allows the replacement of the SystemHelper singleton
+// with a MockSystemHelper.
+class SystemHelperPeer {
+public:
+  class Handle;
```

Comment:
Just a c++ question but do we need this line?  Since we're defining the class right below it

Reply:
Yes but we have `class Handle { ... }` right below it.  I tested locally and we could remove this line if we move the `replaceSystemHelper` definition below the `Handle` definition.  

I was just wondering if it's a style choice or leftover code from an earlier version.

Reply:
Correct, if you moved the replaceSystemHelper() line below the Handle definition, you wouldn't need the forward declaration b/c the compiler has already seen it.

Ill let Ryan comment on why he chose this way